### ISSUE TITLE
TC-Gen: Create a 0.1 degree lat/lon data file for hurricane basins.

### ISSUE THREAD

**Issue by JohnHalleyGotway** [dtcenter/MET#1274]
This task was split off from the original TC-Gen development issue:
https://github.com/NCAR/MET/issues/1127

TC-Genesis forecasts include a lat/lon but do not specify the basin to which that lat/lon belongs.  Therefore, filtering by basin (e.g. basin = [ "AL" ];) is not useful.  Instead, we need to filter by masking region.  However, we should really provide a standard definition of the basins by creating a global NetCDF 0.1 degree lat/lon file.  Store an integer for each grid point that is a flag value defining the hurricane basin to which that belongs.

**Comment by JohnHalleyGotway** [comment]
The results shown below include the changes from Jonathan Vigh on 03/11/2020.
The results shown below include changes from the HWRF group (Evan and Kathryn) on 03/12/2020 requesting that the southern hemisphere basins be extended south to 60 degrees.
Corrected La Reunion to extend south to 40S (not 30) and the western edge of New Delhi as 45E (not 40) on 03/12/2020.

See attached commands for generating these regions:
[gen_rsmc_basins.txt](https://github.com/NCAR/MET/files/4327948/gen_rsmc_basins.txt)

Region definitions:
[poly.tar.gz](https://github.com/NCAR/MET/files/4327951/poly.tar.gz)

Resulting RSMC regions:
![rsmc_global_tenth_degree](https://user-images.githubusercontent.com/21087144/76591324-a64ace80-64b5-11ea-8e97-598503441ddb.png)

Resulting BASIN regions:
![basin_global_tenth_degree](https://user-images.githubusercontent.com/21087144/76591337-aea30980-64b5-11ea-9d51-9b5496e13e83.png)

Combined the resulting variables together in a NetCDF file named:
basin_global_tenth_degree.nc
netcdf basin_global_tenth_degree {
dimensions:
        lat = 1801 ;
        lon = 3601 ;
        mxstr = 24 ;
        nrmsc = 9 ;
        nbasin = 9 ;
variables:
        float lat(lat) ;
                lat:long_name = "latitude" ;
                lat:units = "degrees_north" ;
                lat:standard_name = "latitude" ;
        float lon(lon) ;
                lon:long_name = "longitude" ;
                lon:units = "degrees_east" ;
                lon:standard_name = "longitude" ;
        float basin(lat, lon) ;
                basin:long_name = "Tropical Cyclone Basin Value" ;
                basin:_FillValue = -9999.f ;
        char basin_name(nbasin, mxstr) ;
                basin_name:long_name = "Tropical Cyclone Basins" ;
        float rsmc(lat, lon) ;
                rsmc:long_name = "RSMC Value" ;
                rsmc:_FillValue = -9999.f ;
        char rsmc_name(nrmsc, mxstr) ;
                rsmc_name:long_name = "Regional Specialized Meteorological Centres or Tropical Cyclone Warning Centres" ;

// global attributes:
                :MET_version = "V9.0" ;
                :MET_tool = "gen_vx_mask" ;
                :Projection = "LatLon" ;
                :lat_ll = "-90.000000 degrees_north" ;
                :lon_ll = "-180.000000 degrees_east" ;
                :delta_lat = "0.100000 degrees" ;
                :delta_lon = "0.100000 degrees" ;
                :Nlat = "1801 grid_points" ;
                :Nlon = "3601 grid_points" ;

**Comment by JohnHalleyGotway** [comment]
References for RMSC and BASIN definitions from Jonathan Vigh:

Below, please find the relevant information and source info (in a format that you can stick in the documentation or copy into the GitHub issue), followed by answers to your specific questions.

Key for information provided: WMO region (responsible organization - basins), location of document where Area of Responsibility information can be found (followed by the link to the document and any relevant excerpts of graphics).

Region I (RSMC La Reunion - South-West Indian Ocean), page 1 of ANNEX: 
https://www.wmo.int/pages/prog/www/tcp/documents/OperationalPlan_TCP-12_WMO-1178_2016_En.pdf  
"The area of responsibility of the RSMC includes the tropical and subtropical areas of the South-West Indian Ocean from the Equator to 40°S and west of 90°E to Africa (therefore including the Mozambique Channel)."   

Region II (RSMC New Delhi - Bay of Bengal and Arabian Sea), page II-9 of: 
https://www.wmo.int/pages/prog/www/tcp/documents/PTC-OperationalPlan_TCP-21_Edition2018.pdf  
"The area of responsibility for the issue of tropical cyclone advisories by RSMC Tropical Cyclones, New Delhi cover sea areas of north Indian Ocean between long. 45° E to 100° E."

Region III (RSMC Tokyo - Western Pacific): I could not find any basin definition information in their document, but I think 100° E to 180° E should suffice.
https://www.wmo.int/pages/prog/www/tcp/documents/TCP-23EDITION2019.pdf  

Region IV (RSMC Miami - Atlantic / Eastern Pacific), page 2-1 of: 
https://www.wmo.int/pages/prog/www/tcp/documents/RA-IV-HC-41-OperationalPlan_en_final.pdf
"The area of responsibility of RSMC Miami for issuing tropical and subtropical cyclone advisories is the North Atlantic Ocean, the Caribbean Sea, Gulf of Mexico, North Pacific Ocean eastward from 140ºW."

Region V: (RSMC Nadi and a variety of Tropical Cyclone Warning Centers (TCWCs), as listed below - Southwest Pacic and South-West Indian Ocean), page II-2 of: 
https://www.wmo.int/pages/prog/www/tcp/documents/RAV_OpPlan_TCP24_2018_Final.pdf 
 
![australia](https://user-images.githubusercontent.com/21087144/76544021-f8a3d500-644c-11ea-8a6a-e469846743aa.png)

Description of sub-regions.
![australia_tcwc](https://user-images.githubusercontent.com/21087144/76544102-15d8a380-644d-11ea-8dfa-45d37884f47a.png)

In summary, here are the answers to your specific questions:

- Is the boundary between La Reunion and Australia really LON=90E? (Looks more like 88)  Yes, LON=90E.
- Is the southwest latitude of Australia really LAT=38S?   No, LAT=36S
- Is the southeast latitude of Australia really LAT=35S?    No, LAT=32S
- Is the boundary between Nadi and Wellington really LON = 25S?   Yes

**Comment by JohnHalleyGotway** [comment]
Final version of this file was committed to the develop branch and unit_tc_gen.xml uses this data to filter genesis by basin.
